src/main/java/com/marklogic/kafka/connect/source/CsvPlanInvoker.java

```diff
@@ -29,9 +33,11 @@ public Results invokePlan(PlanBuilder.Plan plan, String topic) {
         if (result.get() != null) {
             try (BufferedReader reader = new BufferedReader(new StringReader(result.get()))) {
                 String headers = reader.readLine();
+                AtomicLong rowNumber = new AtomicLong(1);
```

Comment:
Java is annoying here, and the use of AtomicLong is more complex than it should be - I think it's better to convert the "forEach" call to just be a regular for loop, and then use a regular "int rowNumber". The performance difference between "rowNumber++" and "rowNumber.getAndIncrement()" is likely very close to zero - it's more the simplicity that I'm looking for.

Reply:
On second thought - this is fine, can leave as is.